### ISSUE TITLE
Having multiple domains for the same app, with only one server daemon running

### ISSUE THREAD

**Issue by czj** [typicode/hotel#165]
Hello ! Before everything else, thanks a lot for this great piece of software :)

We have a Rails app that runs on several TLD at the same time like :

```
www.myapp.fr
www.myapp.com
www.myapp.es
```

We can add the app 3 times separately to have it running locally, but it will start a daemon for each. The problem is Puma (Ruby HTTP server) does not like running many times in the same app directory :

```
sh -c rails s -p $PORT

=> Booting Puma
=> Rails 5.0.1 application starting in development on http://localhost:53191
=> Run `rails server -h` for more startup options
A server is already running. Check /Users/the_user/github/myapp/tmp/pids/server.pid.
Exiting
```

We could ask it to use a pid file for each instance of the app, but it would be much simpler to have some sort of domain / TLD alias in the JSON config file, maybe ? As you already have wildcard matching for subdomains, I wonder if this could this be added to hotel ?

Thanks a lot !

**Comment by robinvdvleuten** [comment]
This would be awesome! @czj did you manage to get it working with hotel?

**Comment by czj** [comment]
I did not, unfortunately. Maybe Hotel evolved in the last 4 months, and I cloud check again.

I've been going back to puma-dev for Ruby apps, and thinking about moving to the most reliable solution, yet most complicated : Docker.

**Comment by robinvdvleuten** [comment]
Yeah docker and a good host config app is maybe a better setup for having multiple top-level domains for your development environment.

**Comment by rafapolo** [comment]
you are misunderstanding how the internet works; If the app is the same for the multiple domains, it is just a question of pointing the domains DNS A values to the same puma server IP, exposed at 80. it is a DNS configuration, not the http server. ;)

**Comment by czj** [comment]
I'm not using Hotel on a public IP or a DNS based domain name. I'm using it in development mode, on macOS, with (if I'm correct) Hotel being the resolver. Then I'm using several local dev domains to simulate the public domains, like :

```
www.myapp.fr.dev
www.myapp.com.dev
www.myapp.es.dev
```

If you use Pow ( www.pow.cx ) or Puma-dev ( https://github.com/puma/puma-dev ) which use a symbolic link based configuration in `~/pow/`for example, they allow you to define many domains that point to the same running instance of an app (the same directory).

That's what I did not manage to do with Hotel and would love to be able to achieve :)

**Comment by rafapolo** [comment]
Sorry that I ended up here looking for other puma configuration problem... not sure about hotel. 
Why dont you just add it to /etc/hosts ? (for non-windows)
127.0.0.1 domain.fr.dev
127.0.0.1 domain.es.dev
(dont solve hotel lack of feature, but remove a layer of complexity)

**Comment by czj** [comment]
I'm not sure it will work as the resolver has to be aware of the port
translation he has to do for some domains ? I could try, indeed.

We have applications that use wildcard subdomains, some others that run on
several TLDs and some that use both features. Hard to do that in a hosts
file, but we manage it with the other tools (that only run Ruby processes)

**Comment by Ti-webdev** [comment]
Feature request is supporting multiple aliases to same app
foo.json:
```json
{
  "aliases": ["bar", "baz"],
  "cwd": "/home/...",
  "cmd": "///",
  "env": {}
}
```
to allow open foo.localhost, bar.localhost and baz.localhost.

I want same feature.
In my case I have subdomains (and its not problem with hotel) and totally different names to same app.
